packages/sit-onyx/src/components/OnyxDataGrid/features/filtering/types.ts

```diff
@@ -0,0 +1,71 @@
+import type { MaybeRefOrGetter } from "vue";
+import type { DataGridEntry } from "../../types";
+import type { Compare } from "../all";
+
+/**
+ * Configuration for filtering on a per-column basis.
+ * If at least one column has a filtering configuration, filtering must be explicitly enabled for all columns.
+ */
+export type FilterColumnOptions<TEntry extends DataGridEntry> = {
+  [TKey in keyof TEntry]?: {
+    /**
+     * Determines if filtering is enabled for this column.
+     * If set to true, the column will be filterable.
+     */
+    enabled: boolean;
+    /**
+     * The filter value for this column.
+     * This value will be used to filter the data in this column.
+     */
+    filter?: string;
+    /**
+     * Configuration for how filtering should behave for this column.
+     */
+    filterConfig?: FilterConfig;
+    /**
+     * A custom filtering function for this column.
+     * This function is used to filter the column data, instead of using the default filtering behavior.
+     */
+    filterFunc?: Compare<TEntry[TKey]>;
```

Comment:
Please see #1941 for how this function should look like. Compare is used for sorting, not for filtering :)